src/graph/graphcontext.c

```diff
@@ -62,47 +76,60 @@ static GraphContext *_GraphContext_Create(RedisModuleCtx *ctx, const char *graph
 	return gc;
 }
 
-GraphContext *GraphContext_Retrieve(RedisModuleCtx *ctx, const char *graphName, bool readOnly,
+GraphContext *GraphContext_Retrieve(RedisModuleCtx *ctx, RedisModuleString *graphID, bool readOnly,
 									bool shouldCreate) {
-	assert(graphName);
-
 	GraphContext *gc = NULL;
 	int rwFlag = readOnly ? REDISMODULE_READ : REDISMODULE_WRITE;
-	RedisModuleString *graphID = RedisModule_CreateString(ctx, graphName, strlen(graphName));
 
 	RedisModuleKey *key = RedisModule_OpenKey(ctx, graphID, rwFlag);
 	if(RedisModule_KeyType(key) == REDISMODULE_KEYTYPE_EMPTY) {
 		// Key doesn't exists, create it.
 		if(shouldCreate) {
+			const char *graphName = RedisModule_StringPtrLen(graphID, NULL);
 			gc = _GraphContext_Create(ctx, graphName, GRAPH_DEFAULT_NODE_CAP, GRAPH_DEFAULT_EDGE_CAP);
 		}
 	} else if(RedisModule_ModuleTypeGetType(key) == GraphContextRedisModuleType) {
 		gc = RedisModule_ModuleTypeGetValue(key);
 	}
 	RedisModule_CloseKey(key);
 
-	RedisModule_FreeString(ctx, graphID);
-	if(gc) Graph_SetMatrixPolicy(gc->g, SYNC_AND_MINIMIZE_SPACE);
+	if(gc) {
+		_GraphContext_IncreaseRefCoun(gc);
+		Graph_SetMatrixPolicy(gc->g, SYNC_AND_MINIMIZE_SPACE);
+	}
 	return gc;
 }
 
-void GraphContext_Delete(GraphContext *gc, RedisModuleCtx *ctx) {
-	assert(gc && ctx);
+void GraphContext_Release(GraphContext *gc) {
+	assert(gc);
+	_GraphContext_DecreaseRefCoun(gc);
+}
 
-	const char *graph_name = gc->graph_name;
+void GraphContext_MarkWriter(RedisModuleCtx *ctx, GraphContext *gc) {
```

Comment:
Wow, I wish there was a module `touch` call. But the logic here all looks sensible to me!

The callers all have the required `RedisModuleString graphID` from their argv, which could be passed as an argument instead of building and freeing it here. Fine to not do that if you think it muddles the interface, though.

Reply:
Although the redismodulestring is available through argv,
I don't want to dig into it and retrieve it, in addition I believe the interface is cleaner this way and the overhead is really negligible.